beacon_node/client/src/builder.rs

```diff
@@ -338,8 +339,8 @@ where
                 };
 
                 debug!(context.log(), "Downloading finalized block");
-                // Find a suitable finalized block on an epoch boundary.
-                let mut block = remote
+                // Find a suitable finalized block.
+                let block = remote
                     .get_beacon_blocks_ssz::<TEthSpec>(BlockId::Finalized, &spec)
                     .await
                     .map_err(|e| match e {
```

Comment:
I think we should download the `finalized` state first, then the block from the corresponding slot. Otherwise there could be a race between downloading the state and it being pruned from the remote server, i.e. by the time we request the state by slot it's already gone. If we get the state first, then request the block by slot, we don't have this problem because blocks don't get pruned.

Reply:
great call!

Reply:
Sounds good. We could also read the latest block's slot from `state.latest_block_header`